ir/v1.cpp

```diff
@@ -130,6 +130,19 @@ unsigned IR::Primitive::inferOperandTypes() const {
     return 0;
 }
 
+// infer the index width of a meter/counter/register based on the instance count
+// default to 32 bits if we can't figure it out
+int IR::Stateful::index_width() const {
+    return instance_count > 1 ? ceil_log2(instance_count) : 32;
```

Comment:
A minor edge case, but if someone creates an object with size=1, should the index_width also be 1?

Reply:
Possibly -- there's no good answer.  There really should not be any index in that case, but we can't use `bit<0>` and don't have a good way of having it specialized